src/commands/updateContainer/updateContainer.ts

```diff
@@ -0,0 +1,70 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { type Revision } from "@azure/arm-appcontainers";
+import { AzureWizard, createSubscriptionContext, type IActionContext, type ISubscriptionContext } from "@microsoft/vscode-azext-utils";
+import { type ContainerAppModel } from "../../tree/ContainerAppItem";
+import { type ContainerItem } from "../../tree/containers/ContainerItem";
+import { ContainersItem } from "../../tree/containers/ContainersItem";
+import { createActivityContext } from "../../utils/activityUtils";
+import { getManagedEnvironmentFromContainerApp } from "../../utils/getResourceUtils";
+import { getVerifyProvidersStep } from "../../utils/getVerifyProvidersStep";
+import { localize } from "../../utils/localize";
+import { pickContainer } from "../../utils/pickItem/pickContainer";
+import { getParentResourceFromItem, isTemplateItemEditable, throwTemplateItemNotEditable } from "../../utils/revisionDraftUtils";
+import { ImageSourceListStep } from "../image/imageSource/ImageSourceListStep";
+import { RevisionDraftDeployPromptStep } from "../revisionDraft/RevisionDraftDeployPromptStep";
+import { type ContainerUpdateContext } from "./ContainerUpdateContext";
+import { ContainerUpdateDraftStep } from "./ContainerUpdateDraftStep";
+import { RegistryAndSecretsUpdateStep } from "./RegistryAndSecretsUpdateStep";
+
+// Updates both the 'image' and 'environmentVariables' portion of the container profile
+export async function updateContainer(context: IActionContext, node?: ContainersItem | ContainerItem): Promise<void> {
+    const item: ContainerItem | ContainersItem = node ?? await pickContainer(context, { autoSelectDraft: true });
+    const { containerApp, subscription } = item;
+
+    if (!isTemplateItemEditable(item)) {
+        throwTemplateItemNotEditable(item);
```

Comment:
nit: Very small nit-- kind of weird that this is a function and not just an error type.

Reply:
Yeah that's fair, would you rather I create a function that returns a string and throw that as the error message?  Or don't worry about standardizing the error message?

Reply:
You could do something like:

```
export class TemplateItemNotEditableError extends Error {
    constructor(item: RevisionsItemModel) {
        super(localize('itemNotEditable', 'Action cannot be performed on revision "{0}" because a draft is currently active.', item.revision.name));
    }
}
```

And then `throw new TempalteItemNotEditableError`